src/Shared/Dapr/Core/IDaprSidecarResource.cs

```diff
@@ -8,6 +8,6 @@ namespace CommunityToolkit.Aspire.Hosting.Dapr;
 /// <summary>
 /// Represents a Dapr sidecar resource.
 /// </summary>
-public interface IDaprSidecarResource : IResource
+public interface IDaprSidecarResource : IResource, IResourceWithWaitSupport
```

Comment:
Did you implement the waiting? To make this work, the resource needs a lifecycle, see https://github.com/dotnet/aspire/blob/main/docs/specs/appmodel.md#built-in-resources--lifecycle.

When you implement IResourceWithWaitSupport, at some point you have to raise https://github.com/dotnet/aspire/blob/364382724b4cd3590ec75010525ce36ef0b5e769/src/Aspire.Hosting/ConnectionStringBuilderExtensions.cs#L92